mathjax3-ts/output/chtml/Wrappers/mtable.ts

```diff
@@ -0,0 +1,387 @@
+        let i = 0;
+        for (const row of this.childNodes) {
+            const align = (row.node.attributes.get('rowalign') as string) || rowAlign[i++];
+            if (align !== 'baseline') {
+                row.chtml.style.verticalAlign = align;
+            }
+            for (const cell of row.childNodes) {
+                const calign = cell.node.attributes.get('rowalign') as string;
+                if (calign && calign !== align) {
+                    cell.chtml.style.verticalAlign = calign;
+                }
+            }
+        }
+    }
+
+    /*
+     * Set the inter-row spacing for all rows
+     *  (Use frame spacing on the outsides, if needed, and use half the row spacing on each
+     *   neighboring row, so that if row lines are needed, they fall in the middle
+     *   of the row space.)
+     *
+     * @param{boolean} frame  Whether to include frame spacing on the top and bottom or not
+     * @param{string} fspace  The frame spacing to use, if any
+     */
+    protected handleRowSpacing(frame: boolean, fspacing: string) {
+        //
+        //  Get the row spacing values, and add the frame spacing values at the left and right
+        //
+        const spacing = this.convertLengths(this.getRowAttributes('rowspacing'), 2);
+        if (!spacing) return;
+        spacing.unshift(fspacing);
+        spacing[this.numRows] = fspacing;
+        //
+        //  For each row...
+        //
+        let i = 0;
+        for (const row of this.childNodes) {
+            //
+            //  Get the top and bottom spacing
+            //
+            const tspace = spacing[i++];
+            const bspace = spacing[i];
+            //
+            //  For each cell in the row...
+            //
+            for (const cell of row.childNodes) {
+                //
+                //  Set the style for the spacing, if it is needed, and isn't the
+                //  default already set in the mtd styles
+                //
+                const style = cell.chtml.style;
+                if ((i > 1 || frame) && tspace !== '.125em') {
+                    style.paddingTop = tspace;
+                }
+                if ((i < this.numRows || frame) && bspace !== '.125em') {
+                    style.paddingBottom = bspace;
+                }
+            }
+        }
+    }
+
+    /*
+     * Add borders to the tops of cells to make the row lines
+     */
+    protected handleRowLines() {
+        if (this.node.attributes.get('rowlines') === 'none') return;
+        const lines = this.getRowAttributes('rowlines');
+        if (!lines) return;
+        let i = 0;
+        for (const row of this.childNodes.slice(1)) {
+            const line = lines[i++];
+            if (line === 'none') continue;
+            for (const cell of Array.from(row.chtml.childNodes) as HTMLElement[]) {
+                cell.style.borderTop = '.07em ' + line;
+            }
+        }
+    }
+
+    /*
+     * Add a frame to the mjx-itable, if needed
```

Comment:
Typo: mjx-table?

Reply:
No, mjx-itable is styled as `display: inline-table`, and that is the `firstChild` being modified.  I have changed the comment "add a frame to the table, if needed".